dffml/secrets/from_file.py

```diff
@@ -0,0 +1,51 @@
+import os
+import hashlib
+import asyncio
+import pathlib
+from typing import Union
+
+from ..base import config
+from ..util.entrypoint import entrypoint
+from .base import BaseSecretsContext,BaseSecrets
+
+@config
+class FileSecretsConfig:
+    filename: str
+
+class FileSecretsContext(BaseSecretsContext):
```

Comment:
I would say you should probably just subclass from `INISourceContext`

Reply:
Sorry, this should be a subclass of `MemorySourceContext`, and `FileSecrets` should be a subclass of `INISource`, that way you can define `set` and `get` methods which just grab from `self.mem`. I'd recommend splitting on `.` and using the zeroith index as the key into `self.mem` if the split results in a single element list, then use a key of `__global__` or something.